### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by daven-gomes-cko**
## Proposed changes

Merging a NetworkError bugfix.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [x] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [x] Lint and unit tests pass locally with my changes
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Source/CheckoutAPIClient.swift [hunk 1]
```diff
@@ -107,12 +108,48 @@ public class CheckoutAPIClient {
...
+    /// - parameter card: Card used to create the token
+    /// - parameter completion: Callback to execute if the request is successful or failed
+    public func createCardToken(
+        card: CkoCardTokenRequest,
+        completion: @escaping ((Swift.Result<CkoCardTokenResponse, NetworkError>) -> Void)
+    ) {
+        let url = "\(environment.urlPaymentApi)\(Endpoint.tokens.rawValue)"
+        let jsonEncoder = JSONEncoder()
+        // swiftlint:disable:next force_try
+        var urlRequest = try! URLRequest(url: URL(string: url)!, method: HTTPMethod.post, headers: headers)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:123]: please get rid of forced unwrapping here.

File: Source/CheckoutAPIClient.swift [hunk 2]
```diff
@@ -144,5 +181,41 @@ public class CheckoutAPIClient {
...
+    ///
+    /// - parameter paymentData: Apple Pay payment data used to create a card token
+    /// - parameter completion: Callback to execute if the request is successful or failed
+    public func createApplePayToken(
+        paymentData: Data,
+        completion: @escaping ((Swift.Result<CkoCardTokenResponse, NetworkError>) -> Void)
+    ) {
+        let url = "\(environment.urlPaymentApi)\(Endpoint.tokens.rawValue)"
+        // swiftlint:disable:next force_try
+        var urlRequest = try! URLRequest(url: URL(string: url)!, method: HTTPMethod.post, headers: headers)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:195]: please get rid of forced unwrapping here.

File: Source/CheckoutAPIClient.swift [hunk 3]
```diff
@@ -144,5 +181,41 @@ public class CheckoutAPIClient {
...
+            case .success:
+                do {
+                    let applePayToken = try decoder.decode(CkoCardTokenResponse.self, from: response.data!)
+                    completion(.success(applePayToken))
+                } catch let error {
+                    completion(.failure(.other(error: error)))
+                }
+            case .failure(let responseError):
+                do {
+                    let networkError = try decoder.decode(NetworkError.self, from: response.data!)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:212]: please get rid of forced unwrapping here.

File: Source/CheckoutAPIClient.swift [hunk 4]
```diff
@@ -144,5 +181,41 @@ public class CheckoutAPIClient {
...
+        let applePayTokenData = try? JSONDecoder().decode(ApplePayTokenData.self, from: paymentData)
+        let applePayTokenRequest = ApplePayTokenRequest(token_data: applePayTokenData)
+        urlRequest.httpBody = try? JSONEncoder().encode(applePayTokenRequest)
+
+        request(urlRequest).validate().responseJSON { response in
+            let decoder = JSONDecoder()
+            switch response.result {
+            case .success:
+                do {
+                    let applePayToken = try decoder.decode(CkoCardTokenResponse.self, from: response.data!)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:205]: please get rid of forced unwrapping here.

File: Source/CardViewController.swift [hunk 5]
```diff
@@ -196,11 +196,15 @@ public class CardViewController: UIViewController,
                                        phone: billingDetailsPhone)
         if let checkoutApiClientUnwrap = checkoutApiClient {
```
Comment (deepesh-vasthimal-cko) [inline Source/CardViewController.swift:197]: optional just naming : but you can instead `if let checkoutApiClient = checkoutApiClient {` adding Unwrap is not required.

**Review by deepesh-vasthimal-cko — APPROVED** [review]

**Review by nicolas-maalouf-cko — APPROVED** [review]

**Review by ioan-ghisoi-cko — APPROVED** [review]

**Pull request merged**